src/main/java/io/electrum/prepaidutility/api/EventsResource.java

```diff
@@ -0,0 +1,77 @@
+package io.electrum.prepaidutility.api;
+
+import io.electrum.prepaidutility.model.ErrorDetail;
+import io.electrum.prepaidutility.model.PurchaseResponse;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.Authorization;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+
+@Path(PrepaidUtilityApi.API_BASE_PATH)
+@Api(description = "the events API", authorizations = { @Authorization("httpBasic") })
+public abstract class EventsResource {
+
+   protected abstract IEventsResource getResourceImplementation();
+
+   public static final String PATH = "/events";
+
+   public class NotifyTokenPurchase {
+      public static final String NOTIFY_TOKEN_PURCHASE = "notifyTokenPurchase";
+      public static final int SUCCESS = 202;
+      public static final String PATH = EventsResource.PATH + "/tokenPurchases/{" + PathParameters.PURCHASE_ID + "}";

```

Comment:
because all the operations for this resource start with `EventsResource.PATH` we've tended to create `FULL_PATH` and `RELATIVE_PATH` fields:
```suggestion
      public static final String RELATIVE_PATH = "/tokenPurchases/{" + PathParameters.PURCHASE_ID + "}";
      public static final String FULL_PATH = EventsResource.PATH + RELATIVE_PATH;
```

Reply:
I don't really understand this pattern. When I look at airtime, for example, the FULL_PATH variable never gets used. What is its purpose?

Also, I don't see the FULL_PATH pattern used for other resources in ppu - it's another of those "to refactor or not to refactor" head-scratchers. I'm of the opinion that it is better to just match the existing pattern for a small addition like this.